pkg/verifier/gcp/gcp_verifier.go

```diff
@@ -145,57 +151,64 @@ func (g *GcpVerifier) createComputeServiceInstance(input createComputeServiceIns
 
 }
 
-func (g *GcpVerifier) findUnreachableEndpoints(projectID, zone, instanceName string) error {
+func (g *GcpVerifier) findUnreachableEndpoints(projectID, zone, instanceName string, probe probes.Probe) error {
+
 	// Compile the regular expressions once
-	reVerify := regexp.MustCompile(userdataEndVerifier)
-	reUnreachableErrors := regexp.MustCompile(`Unable to reach (\S+)`)
+	// reVerify := regexp.MustCompile(userdataEndVerifier)
+	// reUnreachableErrors := regexp.MustCompile(`Unable to reach (\S+)`)
 
-	// getConsoleOutput then parse, use c.output to store result of the execution
+	var consoleOutput string
+	g.Logger.Debug(context.TODO(), "Scraping console output and waiting for user data script to complete...")
+
+	// Scrapes console at specified interval up to specified timeout
 	err := helpers.PollImmediate(30*time.Second, 4*time.Minute, func() (bool, error) {
+		// Get the console output from the ComputeService instance
 		output, err := g.GcpClient.GetInstancePorts(projectID, zone, instanceName)
 		if err != nil {
 			return false, err
 		}
 
-		if output != nil {
-			// First, gather the ComputeService console output
-			scriptOutput := fmt.Sprintf("%#v", output)
-			if err != nil {
-				// unable to decode output. we will try again
-				g.Logger.Debug(context.TODO(), "Error while collecting console output, will retry on next check interval: %s", err)
-				return false, nil
-			}
-
-			// In the early stages, an ComputeService instance may be running but the console is not populated with any data, retry if that is the case
-			if len(scriptOutput) < 1 {
-				g.Logger.Debug(context.TODO(), "ComputeService console output not yet populated with data, continuing to wait...")
-				return false, nil
-			}
+		if output == nil {
+			return false, nil
+		}
 
-			// Check for the specific string we output in the generated userdata file at the end to verify the userdata script has run
-			// It is possible we get EC2 console output, but the userdata script has not yet completed.
-			verifyMatch := reVerify.FindString(string(scriptOutput))
-			if len(verifyMatch) < 1 {
-				g.Logger.Debug(context.TODO(), "ComputeService console output contains data, but end of userdata script not seen, continuing to wait...")
+		if len(output.Contents) == 0 {
+			g.Logger.Debug(context.TODO(), "ComputeService console output not yet populated with data, continuing to wait...")
+			return false, nil
+		}
+		consoleOutput = string(output.Contents)
```

Comment:
output.Contents is already a string, so we can remove the boxing here.
```suggestion
		consoleOutput = output.Contents
```